src/patient-registration/patient-registration.component.tsx

```diff
@@ -399,6 +409,7 @@ export const PatientRegistration: React.FC = () => {
           const results = Promise.all(requests);
           results.then(response => {}).catch(err => {});
 
+          window.removeEventListener('single-spa:before-routing-event', cancelNavFn);
```

Comment:
@FlorianRappl yea, went with your suggestion and removed the listener upon submission. Unfortunately, had to elevate a lot of state logic from the before save prompt to the parent level